From cae058906568418e33eb6a80a3da457f49795ef9 Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michael@niedermayer.cc>
Date: Thu, 15 Oct 2026 04:58:55 +0200
Subject: [PATCH] avformat/idroqdec: continue after the declared size of the
 first RoQ_INFO chunk

Stream desynchronization Replicated through UnModified FFmpeg

Found during review of the fixes for the security report yyLBi9GsHV9w
Signed-off-by: Michael Niedermayer <michael@niedermayer.cc>
(cherry picked from commit cae058906568418e33eb6a80a3da457f49795ef9)
---
 libavformat/idroqdec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavformat/idroqdec.c b/libavformat/idroqdec.c
index ad904b42b46..c67724b128a 100644
--- a/libavformat/idroqdec.c
+++ b/libavformat/idroqdec.c
@@ -139,7 +139,7 @@ static int roq_read_packet(AVFormatContext *s,
                     return ret;
                 st->codecpar->width  = roq->width  = AV_RL16(preamble);
                 st->codecpar->height = roq->height = AV_RL16(preamble + 2);
-                break;
+                chunk_size -= RoQ_CHUNK_PREAMBLE_SIZE;
             }
             /* don't care about this chunk anymore */
             avio_skip(pb, chunk_size);
-- 
2.53.0

